### PR TITLE
feat: revamp-nanos-statistics-ui

### PR DESCRIPTION
**Pull request by giansb**
### Acceptance Criteria
- This PR includes the revamp of the following screens:
  - Decode TX (Nanos).
  - Push TX (Nanos).
  - DAG (Nanos).
  - Features (Nanos).
  -Statistics.

### Security Checklist
- [ ] Ensure that no new dependencies were added unless strictly necessary, and no development dependencies were included as production ones, as this could increase the risk of dependency hijacking.

### CODE REVIEW

File: src/components/tx/TxTextInput.js [hunk 1]
```diff
@@ -9,7 +9,34 @@ import React from 'react';
 import { Link } from 'react-router-dom';
 
 class TxTextInput extends React.Component {
-  render() {
```
Comment (r4mmer) [inline src/components/tx/TxTextInput.js:12]: These changes are also part of the other PR

File: src/newUi.scss [hunk 2]
```diff
@@ -942,6 +942,24 @@ nav {
   margin: 16px 0;
 }
 
+.alert-success-container-big {
```
Comment (r4mmer) [inline src/newUi.scss:945]: These changes are also part of the other PR

File: src/screens/Dag.js [hunk 3]
```diff
@@ -8,6 +8,7 @@
 import React, { useCallback, useEffect, useRef, useState } from 'react';
 import txApi from '../api/txApi';
 import DagComponent from '../components/DagComponent';
+import { useNewUiEnabled } from '../hooks';
```
Comment (r4mmer) [inline src/screens/Dag.js:11]: These changes are also part of the other PR

File: src/screens/DecodeTx.js [hunk 4]
```diff
@@ -5,11 +5,14 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-import React, { useState } from 'react';
```
Comment (r4mmer) [inline src/screens/DecodeTx.js:8]: These changes are also part of the other PR

File: src/screens/FeatureList.js [hunk 5]
```diff
@@ -7,13 +7,28 @@
 
 import React from 'react';
 import Features from '../components/feature_activation/Features';
+import { useNewUiEnabled } from '../hooks';
```
Comment (r4mmer) [inline src/screens/FeatureList.js:10]: These changes are also part of the other PR

File: src/screens/PushTx.js [hunk 6]
```diff
@@ -8,6 +8,9 @@
 import React, { useRef, useState } from 'react';
 import TxTextInput from '../components/tx/TxTextInput';
 import txApi from '../api/txApi';
+import { useNewUiEnabled } from '../hooks';
```
Comment (r4mmer) [inline src/screens/PushTx.js:11]: These changes are also part of the other PR

**Comment by tuliomir** [comment]
@giansb , would it be possible for you to change the base branch of this PR to the branch used by #345 ?
It's not viable for us to review the same code in two different PRs

File: src/components/DagComponent.js [hunk 7]
```diff
@@ -366,6 +370,51 @@ class DagComponent extends React.Component {
       });
   }
 
+  newDrawGraph() {
```
[the remark this answers is not in the document]
Comment (giansb) [reply src/components/DagComponent.js:373]: Yes, it was done, nothing new on this one.🙂

File: src/components/feature_activation/Features.js [hunk 8]
```diff
@@ -15,6 +15,7 @@ import FeatureRow from './FeatureRow';
 import colors from '../../index.scss';
 import PaginationURL from '../../utils/pagination';
 import featureApi from '../../api/featureApi';
+import { DropDetails } from '../DropDetails';
```
[the remark this answers is not in the document]
Comment (giansb) [reply src/components/feature_activation/Features.js]: PR merged with master. Waiting Review.🙂

File: src/components/tx/TxTextInput.js (hunk 1 above)
Comment (giansb) [reply src/components/tx/TxTextInput.js]: PR merged with master. Waiting Review.🙂

File: src/components/tx/TxData.js [hunk 9]
```diff
@@ -5,15 +5,16 @@
  * LICENSE file in the root directory of this source tree.
  */
 
+import $ from 'jquery';
```
[the remark this answers is not in the document]
Comment (giansb) [reply src/components/tx/TxData.js]: Yes, now the PR has been merged with master
. Waiting Review.🙂

File: src/newUi.scss (hunk 2 above)
Comment (giansb) [reply src/newUi.scss]: PR merged with master. Waiting Review.🙂

File: src/screens/Dag.js (hunk 3 above)
Comment (giansb) [reply src/screens/Dag.js]: PR merged with master. Waiting Review.🙂

File: src/screens/DecodeTx.js (hunk 4 above)
Comment (giansb) [reply src/screens/DecodeTx.js]: PR merged with master. Waiting Review.🙂

File: src/screens/FeatureList.js (hunk 5 above)
Comment (giansb) [reply src/screens/FeatureList.js]: PR merged with master. Waiting Review.🙂

File: src/screens/PushTx.js (hunk 6 above)
Comment (giansb) [reply src/screens/PushTx.js]: PR merged with master. Waiting Review.🙂

File: src/components/DagComponent.js [hunk 10]
```diff
@@ -365,6 +365,51 @@ class DagComponent extends React.Component {
...
```
Comment (tuliomir) [inline src/components/DagComponent.js:368]: question: Why did we need to re-implement this method? It was dropped on #345

File: src/components/tx/TxData.js [hunk 11]
```diff
@@ -493,6 +493,29 @@ class TxData extends React.Component {
       return <ul>{v}</ul>;
     };
 
+    const renderNewUiListWithLinks = hashes => {
```
Comment (tuliomir) [inline src/components/tx/TxData.js:496]: chore: This function is called 9 times, and in each call the interface has to decide whether to render the new UI or the old one.

The function by the interface should continue to be `renderListWithLinks()` and, if this function detects the new UI is active, it returns this `renderNewUiListWithLinks()` function instead. This improves centralization and code clarity.

File: src/screens/TimeSeriesDashboard.js [hunk 12]
```diff
@@ -7,19 +7,36 @@
...
-      <TimeSeries featureFlag={featureFlag} maintenanceMode={maintenanceMode} />
+      <TimeSeries
+        featureFlag={featureFlag}
+        maintenanceMode={maintenanceMode}
+        newUiEnabled={newUiEnabled}
+      />
     </div>
   );
+
+  return newUiEnabled ? renderNewUi() : renderUi();
```
Comment (tuliomir) [inline src/screens/TimeSeriesDashboard.js:39]: question: Is there any difference between the rendering of the new UI and the current one? It seems this decision is not relevant.

**Review by tuliomir — COMMENTED** [review]
Partial review, some tests yet to be executed

File: src/components/DagComponent.js (hunk 10 above)
Comment (giansb) [reply src/components/DagComponent.js]: You are right, it has been corrected. Please review.

File: src/components/tx/TxData.js (hunk 11 above)
Comment (giansb) [reply src/components/tx/TxData.js:478]: Thanks for the suggestion, it has been corrected for clarity. Waiting for review.🙂

File: src/screens/TimeSeriesDashboard.js (hunk 12 above)
Comment (giansb) [reply src/screens/TimeSeriesDashboard.js]: really, there was no difference, it was fixed. Please review.

File: src/components/DagComponent.js [hunk 13]
```diff
@@ -488,14 +488,27 @@ class DagComponent extends React.Component {
...
     return (
       <div ref="dagWrapper" className="dagWrapper">
         <svg id="graph" />
         <div className="tooltip"></div>
       </div>
     );
   }
+
+  render() {
+    return this.props.newUiEnabled ? this.renderNewUi() : this.renderUi();
```
[the remark this answers is not in the document]
Comment (giansb) [reply src/components/DagComponent.js]: You are right, corrected. Waiting for Review.🙂

File: src/components/tx/TxData.js [hunk 14]
```diff
@@ -447,30 +447,12 @@ class TxData extends React.Component {
...
-        }
-        const v = hashes.map(h => (
-          <li key={h}>
-            <Link to={`/transaction/${h}`}>
-              {h} {h === this.props.transaction.hash && ' (Current transaction)'}
-            </Link>
-          </li>
-        ));
-        return <ul>{v}</ul>;
+        // Se a nova interface estiver ativa, delega para renderNewUiListWithLinks
```
[the remark this answers is not in the document]
Comment (giansb) [reply src/components/tx/TxData.js]: Translated into english. Please review.

File: src/components/Sidebar.js [hunk 15]
```diff
@@ -32,6 +32,8 @@ function Sidebar({ close, open }) {
   const theme = useSelector(state => state.theme);
   const sidebarRef = useRef(null);
   const hathorNetwork = `Hathor ${REACT_APP_NETWORK}`;
+  const [tokensOpen, setTokensOpen] = useState(false);
+  const [tollsOpen, setTollsOpen] = useState(false);
```
[the remark this answers is not in the document]
Comment (giansb) [reply src/components/Sidebar.js]: Thanks for the point and suggestion, it has been corrected. Waiting for Review.🙂

File: src/components/tx/TxData.js [hunk 16]
```diff
@@ -558,7 +563,7 @@ class TxData extends React.Component {
               {conflictNotTwin.length > 0 && (
                 <div className="mb-0">
                   <span>Transactions double spending the same outputs as this transaction: </span>
-                  {renderListWithLinks(conflictNotTwin, true)}
+                  {renderListWithLinks(this.props.meta.twins, true)}
```
[the remark this answers is not in the document]
Comment (giansb) [reply src/components/tx/TxData.js]: Hello, probably the first version of renderNewUiListWithLinks used this parameter internally, corrected to maintain the old logic. Awaiting review.

File: src/components/Sidebar.js [hunk 17]
```diff
@@ -90,32 +92,35 @@ function Sidebar({ close, open }) {
...
-                <span className="nav-item dropdown">
+                <span className="dropdown-sidebar">
                   <span
                     className="nav-link dropdown-toggle"
-                    id="navbarDropdown"
-                    role="button"
-                    data-bs-toggle="dropdown"
-                    aria-haspopup="true"
-                    aria-expanded="false"
+                    onClick={() => setTokensOpen(!tokensOpen)}
```
[the remark this answers is not in the document]
Comment (giansb) [reply src/components/Sidebar.js:98]: Corrected style. Please review.🙂

File: src/components/tx/TxData.js [hunk 18]
```diff
@@ -597,7 +603,7 @@ class TxData extends React.Component {
           </h4>
           <div>
             <span>It is voided by: </span>
-            {renderListWithLinks(this.props.meta.voided_by, true)}
+            {renderListWithLinks(conflictNotTwin, true)}
```
Comment (tuliomir) [inline src/components/tx/TxData.js:606]: question: Why have you changed this?

The transactions exhibited are not correct.
image

image

File: src/components/tx/TxData.js [hunk 19]
```diff
@@ -518,14 +523,15 @@ class TxData extends React.Component {
...
+      const conflictNotTwin = this.props.meta.conflict_with.length;
       if (!this.props.meta.twins.length) {
         return null;
       }
       return (
         <div>
           This transaction has twin{' '}
           {helpers.plural(this.props.meta.twins.length, 'transaction', 'transactions')}:{' '}
-          {renderListWithLinks(this.props.meta.twins, true)}
+          {renderListWithLinks(conflictNotTwin, true)}
```
Comment (tuliomir) [inline src/components/tx/TxData.js:534]: question: The parameter was changed from an array to an integer. Why was that needed?

**Review by tuliomir — CHANGES_REQUESTED** [review]

File: src/components/tx/TxData.js (hunk 19 above)
Comment (giansb) [reply src/components/tx/TxData.js]: Ended up being modified along with the other renderListWithLinks by mistake. Fixed. Please review.

File: src/components/tx/TxData.js (hunk 18 above)
Comment (giansb) [reply src/components/tx/TxData.js]: ended up being modified along with the renderListWithLinks of the valid return, but it has already been corrected. Waiting for review.🙂

**Review by tuliomir — DISMISSED** [review]

File: src/utils/useTimelapseCounter.js [hunk 20]
```diff
@@ -0,0 +1,25 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { useEffect, useState } from 'react';
+
+const useTimelapseCounter = initialTimestamp => {
```
Comment (r4mmer) [inline src/utils/useTimelapseCounter.js:10]: suggestion(minor): Move this file to the `hooks` folder.

File: src/utils/useTimelapseCounter.js [hunk 21]
```diff
@@ -0,0 +1,25 @@
...
+  const [renderCount, setRenderCount] = useState(0);
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      const secondsAgo = Math.floor((Date.now() - initialTimestamp) / 1000);
+      setRenderCount(secondsAgo);
+    }, 1000);
```
Comment (r4mmer) [inline src/utils/useTimelapseCounter.js:17]: suggestion(minor): change the timer to 200ms.

This is just to avoid a timing edge case where we miss the timing and the counter takes 2 seconds to update.
If we use 200ms the highest error is 1.2s which is less jarring.

File: src/components/DagComponent.js [hunk 22]
```diff
@@ -488,14 +488,18 @@ class DagComponent extends React.Component {
...
       <div ref="dagWrapper" className="dagWrapper">
         <svg id="graph" />
         <div className="tooltip"></div>
       </div>
     );
   }
+
+  render() {
+    return this.renderUi();
+  }
```
[the remark this answers is not in the document]
Comment (giansb) [reply src/components/DagComponent.js]: Because renderNewUi existed previously but was not needed in this component, it was fixed to keep render with return again. Please review.🙂

File: src/utils/useTimelapseCounter.js (hunk 20 above)
Comment (giansb) [reply src/utils/useTimelapseCounter.js]: Thanks for the suggestion, moved to hooks folder. Waiting review.🙂

File: src/utils/useTimelapseCounter.js (hunk 21 above)
Comment (giansb) [reply src/utils/useTimelapseCounter.js]: Right, timer value changed to 200ms. Please review.🙂

**Review by tuliomir — APPROVED** [review]

**Review by r4mmer — APPROVED** [review]
